public/app/plugins/datasource/cloudwatch/components/MetricsQueryEditor/MetricsQueryEditor.tsx

```diff
@@ -134,18 +135,15 @@ export const MetricsQueryEditor = (props: Props) => {
         {config.featureToggles.cloudWatchDynamicLabels ? (
           <EditorField
             label="Label"
-            width={26}
+            width={52}
             optional
             tooltip="Change time series legend name using Dynamic labels. See documentation for details."
           >
-            <Input
-              id={`${query.refId}-cloudwatch-metric-query-editor-label`}
-              onBlur={onRunQuery}
-              value={preparedQuery.label ?? ''}
-              onChange={(event: ChangeEvent<HTMLInputElement>) =>
-                onChange({ ...preparedQuery, label: event.target.value })
-              }
-            />
+            <DynamicLabelsField
+              onRunQuery={onRunQuery}
+              label={query.label ?? ''}
```

Comment:
```suggestion
               label={preparedQuery.label ?? ''}
```

Then the test should work (see my other comment)

Reply:
I think the test broke because I had to mock the CodeEditor, see https://github.com/grafana/grafana/blob/cloudwatch/dynamic-labels-lang/public/app/plugins/datasource/cloudwatch/components/MetricsQueryEditor/MetricsQueryEditor.test.tsx#L16-L21. The code editor component uses Monaco and unfortunately there's no way to test monaco using jest today.